### ISSUE TITLE
OperationalError: FATAL:  password authentication failed for user "dsmrreader"

### ISSUE THREAD

**Issue by unclekoen** [xirixiz/dsmr-reader-docker#209]
I am trying to install DSMR Reader with Docker Compose using this guide: https://github.com/xirixiz/dsmr-reader-docker#setup

Running it on an Intel Nuc with Proxmox. Docker runs in an Ubuntu Server VM.

I used the exact given example docker-compose.yaml file:
```
version: '3.8'

services:
  dsmrdb:
    # When using Postgres, release 12.x is supported only
    # due to the limited availability of client packages, especially for arm32v7
    image: postgres:13
    restart: always
    volumes:
      - ./dsmrdb:/var/lib/postgresql/data
    environment:
      - TZ=Europe/Amsterdam
      - PG_TZ=Europe/Amsterdam
      - POSTGRES_USER=dsmrreader
      - POSTGRES_PASSWORD=dsmrreader
      - POSTGRES_DB=dsmrreader

  dsmr:
#    build: .
    image: xirixiz/dsmr-reader-docker:latest
    depends_on:
      - dsmrdb
    cap_add:
      - NET_ADMIN
    restart: always
    volumes:
      - ./dsmr_backups:/dsmr/backups
    environment:
      - DJANGO_TIME_ZONE=Europe/Amsterdam
      - VIRTUAL_HOST=localhost
    ports:
      - 7777:80
      - 7779:443
    devices:
      - /dev/ttyUSB0:/dev/ttyUSB0

#volumes:
#   dsmrdb:
#   dsmrdb_backups:
```

Starting goes well with: 
`docker-compose up -d`

But checking the dashboard with:
`http://<hostname>:777`

Gives me the error page:
```
Server Error
Sorry, something unexpected happened.
Exception:
OperationalError: FATAL:  password authentication failed for user "dsmrreader"
Traceback:
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/usr/local/lib/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)

  File "/dsmr/dsmr_frontend/mixins.py", line 10, in dispatch
    if not FrontendSettings.get_solo().always_require_login:

  File "/usr/local/lib/python3.9/site-packages/solo/models.py", line 55, in get_solo
    obj, created = cls.objects.get_or_create(pk=cls.singleton_instance_id)

  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)

  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 573, in get_or_create
    return self.get(**kwargs), False

  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 425, in get
    num = len(clone)

  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 269, in __len__
    self._fetch_all()

  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 1308, in _fetch_all
    self._result_cache = list(self._iterable_class(self))

  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)

  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1154, in execute_sql
    cursor = self.connection.cursor()

  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)

  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 259, in cursor
    return self._cursor()

  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 235, in _cursor
    self.ensure_connection()

  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)

  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 219, in ensure_connection
    self.connect()

  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value

  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 219, in ensure_connection
    self.connect()

  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)

  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 200, in connect
    self.connection = self.get_new_connection(conn_params)

  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)

  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 187, in get_new_connection
    connection = Database.connect(**conn_params)

  File "/usr/local/lib/python3.9/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)

```

I have read same kind of issues, but i have no idea what to do. I think it's easy and i just forgetting something. But please help.

**Comment by xirixiz** [comment]
Can you check ```docker logs -f dsmrdb```? If dsmrdb is running properly then I'd remove the database folder from your host and start over again. But most probably dsmrdb hasn't started properly due to: ```./dsmrdb:/var/lib/postgresql/data```

You could make it work by unchecking:
```
#volumes:
#   dsmrdb:
#   dsmrdb_backups:
```

**Comment by unclekoen** [comment]
Unchecked:
```
#volumes:
#   dsmrdb:
#   dsmrdb_backups:
```

Started over again by creating the new container:
```
ubuntu-user@ubuntuserver:~/dsmrreader$ sudo docker-compose up -d
Creating network "dsmrreader_default" with the default driver
Creating volume "dsmrreader_dsmrdb" with default driver
Creating volume "dsmrreader_dsmrdb_backups" with default driver
Creating dsmrreader_dsmrdb_1 ... done
Creating dsmrreader_dsmr_1   ... done
```


Checking the logs gives me:
```
ubuntu-user@ubuntuserver:~/dsmrreader$ sudo docker logs -f dsmrdb
Error: No such container: dsmrdb
```

**Comment by unclekoen** [comment]
It's creating an other name. So i have got a bit of the logging:
```
ubuntu-user@ubuntuserver:~/dsmrreader$ sudo docker logs -f dsmrreader_dsmrdb_1

PostgreSQL Database directory appears to contain a database; Skipping initialization

2021-06-05 12:43:01.416 CEST [1] FATAL:  could not write lock file "postmaster.pid": No space left on device

PostgreSQL Database directory appears to contain a database; Skipping initialization

2021-06-05 12:43:02.634 CEST [1] FATAL:  could not write lock file "postmaster.pid": No space left on device

PostgreSQL Database directory appears to contain a database; Skipping initialization

2021-06-05 12:43:03.657 CEST [1] LOG:  starting PostgreSQL 13.3 (Debian 13.3-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
2021-06-05 12:43:03.658 CEST [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2021-06-05 12:43:03.659 CEST [1] LOG:  listening on IPv6 address "::", port 5432
2021-06-05 12:43:03.666 CEST [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2021-06-05 12:43:03.675 CEST [25] LOG:  database system shutdown was interrupted; last known up at 2021-06-05 12:41:33 CEST
2021-06-05 12:43:03.875 CEST [25] LOG:  database system was not properly shut down; automatic recovery in progress
2021-06-05 12:43:03.880 CEST [25] LOG:  redo starts at 0/15C7958
2021-06-05 12:43:03.880 CEST [25] LOG:  invalid record length at 0/15C7990: wanted 24, got 0
2021-06-05 12:43:03.880 CEST [25] LOG:  redo done at 0/15C7958
2021-06-05 12:43:03.908 CEST [1] LOG:  database system is ready to accept connections
2021-06-05 12:43:05.998 CEST [33] FATAL:  password authentication failed for user "dsmrreader"
2021-06-05 12:43:05.998 CEST [33] DETAIL:  Role "dsmrreader" does not exist.
        Connection matched pg_hba.conf line 99: "host all all all md5"
2021-06-05 12:43:09.141 CEST [34] FATAL:  password authentication failed for user "dsmrreader"
2021-06-05 12:43:09.141 CEST [34] DETAIL:  Role "dsmrreader" does not exist.
```

**Comment by unclekoen** [comment]
```
ubuntu-user@ubuntuserver:~/dsmrreader$ df -h
Filesystem                         Size  Used Avail Use% Mounted on
udev                               1.9G     0  1.9G   0% /dev
tmpfs                              394M  1.2M  393M   1% /run
/dev/mapper/ubuntu--vg-ubuntu--lv  6.9G  6.5G     0 100% /
tmpfs                              2.0G     0  2.0G   0% /dev/shm
tmpfs                              5.0M     0  5.0M   0% /run/lock
tmpfs                              2.0G     0  2.0G   0% /sys/fs/cgroup
/dev/sda2                          976M  105M  805M  12% /boot
/dev/loop0                          56M   56M     0 100% /snap/core18/1997
/dev/loop1                          70M   70M     0 100% /snap/lxd/19188
/dev/loop2                          33M   33M     0 100% /snap/snapd/11841
/dev/loop3                          56M   56M     0 100% /snap/core18/2066
/dev/loop4                          33M   33M     0 100% /snap/snapd/12057
/dev/loop5                          68M   68M     0 100% /snap/lxd/20326
overlay                            6.9G  6.5G     0 100% /var/lib/docker/overlay2/c6050e39433e242a180b013ae2b2aa23dfcb79738f66bf4fd42f6ca7988b9abd/merged
tmpfs                              394M     0  394M   0% /run/user/1000
overlay                            6.9G  6.5G     0 100% /var/lib/docker/overlay2/a3f3ea9e925c067df312564aee9c3a9f52683ffff6aa949988416feb27d56372/merged

```

**Comment by xirixiz** [comment]
Hmm, it seems you don't have any space left on your device 😄 
You should do some housekeeping.

You can check which volumes are created by Docker:
```docker volume ls``

And to cleanup Docker, you could run something like this (read documentation):
```docker system prune -a -f --volumes```

**Comment by unclekoen** [comment]
I don't get it. It's an VM with 16GB of harddrive. So there should be more then space left.

```
ubuntu-user@ubuntuserver:~$ sudo docker volume ls
DRIVER    VOLUME NAME
local     dsmrreader_dsmrdb
local     dsmrreader_dsmrdb_backups
local     wordpress_db_data
local     wordpress_wp_data
```

```
ubuntu-user@ubuntuserver:~$ sudo docker system df
TYPE            TOTAL     ACTIVE    SIZE      RECLAIMABLE
Images          4         4         1.506GB   69.26MB (4%)
Containers      4         4         22.8MB    0B (0%)
Local Volumes   4         2         394.8MB   0B (0%)
Build Cache     0         0         0B        0B
```

**Comment by xirixiz** [comment]
I think that is something you need to find out yourself. Not sure what's happing why your disks are at 100% usage.
But for sure it's related to the issue you posted.

As far as I can see there is a 6.5GB volume group....

**Comment by xirixiz** [comment]
I`m closing the issue, but ofcourse you're welcome to still ask questions in this ticket.

**Comment by bikeymouse** [comment]
It could be a user-rights issue. 
In previous versions of Postgress I had to change the owner to be the same as the docker-user by adding this line in the docker-compose file:
    user: "1000:1000"
(in your case that would be the id of "ubuntu-user")
With the latest versions of DMSR however I had to remove that again to get it working. 
If I now look at the dsmrdb data directory the owner is "systemd-coredump".
So perhaps changing the owner of the data-directory can help?
